src/Symfony/Component/Security/Guard/Tests/Firewall/GuardAuthenticationListenerTest.php

```diff
@@ -184,6 +204,45 @@ public function testHandleCatchesAuthenticationException()
         $listener->handle($this->event);
     }
 
+    /**
+     * @legacy
+     */
+    public function testLegacyInterfaceAvoidSupportsCall()
+    {
+        $authenticatorA = $this->getMock('Symfony\Component\Security\Guard\GuardAuthenticatorInterface');
+        $providerKey = 'my_firewall3';
+
+        $authenticatorA
+            ->expects($this->never())
+            ->method('supports');
```

Comment:
It tests [this condition](https://github.com/Amo/symfony/blob/master/src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php#L100) is not executed when using the legacy interface